src/Dfe.Spi.UkrlpAdapter.Infrastructure.AzureStorage/Cache/TableStateRepository.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfe.Spi.Common.Logging.Definitions;
+using Dfe.Spi.UkrlpAdapter.Domain.Cache;
+using Dfe.Spi.UkrlpAdapter.Domain.Configuration;
+using Dfe.Spi.UkrlpAdapter.Domain.UkrlpApi;
+using Microsoft.Azure.Cosmos.Table;
+
+namespace Dfe.Spi.UkrlpAdapter.Infrastructure.AzureStorage.Cache
+{
+    public class TableStateRepository : IStateRepository
+    {
+        private readonly ILoggerWrapper _logger;
+        private readonly CloudTable _table;
+
+        public TableStateRepository(CacheConfiguration configuration, ILoggerWrapper logger)
+        {
+            _logger = logger;
+
+            var storageAccount = CloudStorageAccount.Parse(configuration.TableStorageConnectionString);
+            var tableClient = storageAccount.CreateCloudTableClient();
+            _table = tableClient.GetTableReference(configuration.StateTableName);
```

Comment:
Just an observation, doesn't _have to be_ addressed, but you're copying and pasting similar across your table adapters. Why not inherit from a base class and do this in a common/shared constructor?

Reply:
I am not sure, with these bits, it would make things any smaller or clearer. Perhaps if/when there is more things shared it would make sense